seps/sep-0002.md

```diff
@@ -0,0 +1,169 @@
+
+With this approach, can we efficiently determine which group members should receive a notification when a post is created on that group's page?  How challenging would it be to ensure that only group members can post to this group's page? There is a solution here, however these implicitly defined group connections with multiple versions of group membership adds complexity. 
+
+
+1. [Communities on DevHub](https://near.org/devgovgigs.near/widget/Ideas)
+
+DevHub's community implementation exists on a distinct contract from near.social. This results in a lack of discoverability of DevHub's groups across gateways due to its community data not being included in the indexes which power search on socialdB.
+
+
+## Specification 
+Building upon the NFT-based groups implementation, we would benefit from the following the creation and maintanence of two [Core Social Indexers](https://docs.google.com/document/d/14OmYzMYX60lBm2U2Ridz6GZCeN_A6IcU_qn-yPwLTfs/edit#heading=h.nfrlp9tpavvv):
+
+1. A dataplatform.near Core Social indexer for NFT_Groups, that tracks posts to main which include a groupContractId, e.g. writes to post.main.[groupContractId], and provides an API which lists all of the groups which have been created in socialDB. 
+	1. To support the ability to search for groups, where search text is matched against the group's metadata such as name, symbol, description, and spec, the above indexer should fetch and store this metadata from a source such as the [Mintbase API](https://cloud.hasura.io/public/graphiql?endpoint=https%3A%2F%2Fgraph.mintbase.xyz%2Fmainnet&header=mb-api-key%3Aanon&query=query+MyQuery+%7B%0A++nft_metadata%28%0A++++where%3A+%7Breference%3A+%7B_eq%3A+%22nb0-oBR379DzoFYeYv-LesjVNmrVlDs5IqQ8hfDfnMU%22%7D%7D%0A++%29+%7B%0A++++id%0A++++media%0A++++reference_blob%0A++%7D%0A%7D%0A
+
+1. A dataplatform.near Core Social indexer for Group_Members, that maintains a list of NFT owners for each NFT_Group, and provides an API which serves a list of all AccountIds which belong to a group.
+
+
+## Drawbacks
+
+- The NFT-based groups implementation shifts the responsibility of valid NFT management onto users. However, gateways can mitigate this with a Mintbase integration or the ContractWizardUI to faciliate the creation of NFTs.
+
+- The ability to forcibly remove someone from a group, e.g. for violating group guidelines, might not exist - depending on the functionality of the NFT contract for a given group. A potential workaround would be to 'moderate' the user by blocking their posts or banning their account - this would be gateway specific. A gateway could also offer a 'recommended' path for creating a group that includes the creation of an NFT collection with a contract that includes rules that allows the group creator to initiate a transfer to allow the removal of the NFT from any holder.
+
```

Comment:
If there is an ability to remove a user from a group, how should this information be stored in the SocialDB? The indexer needs to know if a group member was removed.

Reply:
@milenakrawczyk what do you recommend?  The schema that the indexer populates could minimally include a key/value pairing of `NFTContractID:[AccountIdList]`

Reply:
Could you explain further? Where would this key/pair be stored?
If we base our approach on NFTs, then we would need to combine the data from NFTs with the key/value pairs you mentioned.

Reply:
Maybe removing a user from a group could be achieved by expiring an NFT as was mentioned by @evgenykuzyakov. We could index specific calls to know if a user is still a member or not.

Reply:
NFTs are not required to have an expiration, it's an optional field. So we cannot depend on leveraging expiration for moderation.

Reply:
> Where would this key/pair be stored?
we need to determine this.